proto/substrait/algebra.proto

```diff
@@ -562,19 +576,7 @@ message HashJoinRel {
   repeated Expression.FieldReference right_keys = 5;
   Expression post_join_filter = 6;
 
-  JoinType type = 7;
-
-  enum JoinType {
-    JOIN_TYPE_UNSPECIFIED = 0;
-    JOIN_TYPE_INNER = 1;
-    JOIN_TYPE_OUTER = 2;
-    JOIN_TYPE_LEFT = 3;
-    JOIN_TYPE_RIGHT = 4;
-    JOIN_TYPE_LEFT_SEMI = 5;
-    JOIN_TYPE_RIGHT_SEMI = 6;
-    JOIN_TYPE_LEFT_ANTI = 7;
-    JOIN_TYPE_RIGHT_ANTI = 8;
-  }
+  PhysicalJoinType type = 7;
```

Comment:
For protobuf compatibility it is better to make this 8 and mark 7 as deprecated.

Reply:
I suppose if we keep all of the values the same as you've done here it would be wire compatible which mean we wouldn't need to renumber the field for the "type" change.

Reply:
Yep, it's still technically wire compatible! The Anti Join will have to default to one of the two options though.

Reply:
Maybe I change this to be 

```
  PHYSICAL_JOIN_TYPE_LEFT_ANTI = 7;
  PHYSICAL_JOIN_TYPE_RIGHT_ANTI = 8;
  PHYSICAL_JOIN_TYPE_LEFT_ANTI_NULL_AWARE = 9;
  PHYSICAL_JOIN_TYPE_RIGHT_ANTI_NULL_AWARE = 10;
```

?

Reply:
Updated to this ^

Reply:
Updated again to use a separate `is_null_aware` field in the rels